src/AEG/ryanTree/ShapeAndPointClasses.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @author Ryan Reilly
+ */
+export class Point {
+    public x: number; public y: number;
+
+    constructor(xPos: number, yPos: number) {
+        this.x = xPos ?? 0; this.y = yPos ?? 0;
+    }
+}
+
+export class Rectangle {
+    public tl: Point; public tr: Point;
+    public bl: Point; public br: Point;

```

Comment:
This is overkill, you just need the tl and br points, everything else can be computed from these.